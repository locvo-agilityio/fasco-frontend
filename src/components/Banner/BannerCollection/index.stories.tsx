import type { Meta, StoryObj } from '@storybook/react';

// Components
import BannerCollection from '.';

const meta = {
  title: 'Components/BannerCollection',
  component: BannerCollection,
  parameters: {
    layout: 'centered',
  },
  render: () => (
    <div className="w-[1920px]">
      <BannerCollection />
    </div>
  ),
  tags: ['autodocs'],
} satisfies Meta<typeof BannerCollection>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
