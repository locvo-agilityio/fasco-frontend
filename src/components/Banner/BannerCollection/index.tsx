'use client';

import Link from 'next/link';
import Image from 'next/image';

// Components
import { Button } from '@/components/common';

// Image
import CollectionImage from '../../../../public/assets/image-collection.webp';

// Constants
import { ROUTES } from '@/constants';

const BannerCollection = () => (
  <div className="relative flex w-full h-full max-h-[1077px] flex-col items-center justify-center bg-background-500">
    <div className="flex w-full h-full">
      <div className="hidden relative md:block w-1/2 h-full min-h-[570px]">
        <Image
          alt="Sign In Image"
          src={CollectionImage}
          quality={100}
          fill
          sizes="50vw"
          priority
        />
      </div>

      <div className="flex flex-col w-full h-full md:w-1/2 items-start justify-center">
        <div className="pt-14 px-14">
          <p className="font-poppins text-gray-700">Women Collection</p>

          <h2 className="text-4xl font-volkhov text-primary-200 mt-5">
            Peaky Blinders
          </h2>

          <p className="font-poppins underline mt-5">DESCRIPTION</p>

          <p className="font-poppins text-gray-600 max-w-[515px] mt-5">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Scelerisque
            duis ultrices sollicitudin aliquam sem. Scelerisque duis ultrices
            sollicitudin. Lorem ipsum dolor sit amet, consectetur adipiscing
            elit. Scelerisque duis.
          </p>

          <div className="mt-5 flex items-center gap-4">
            Size:
            <div className="w-[53px] h-7 bg-black rounded-regular text-white text-center">
              M
            </div>
          </div>

          <p className="font-poppins text-black font-medium mt-5 text-4md">
            $100.00
          </p>

          <Button as={Link} href={ROUTES.SHOP} className="max-w-52 h-14 mt-10">
            Buy Now
          </Button>
        </div>
      </div>
    </div>
  </div>
);

export default BannerCollection;
