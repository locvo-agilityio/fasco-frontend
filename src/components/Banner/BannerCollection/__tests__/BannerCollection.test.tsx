import { render } from '@testing-library/react';

// Components
import BannerCollection from '..';

describe('BannerCollection', () => {
  it('should render correctly', () => {
    const { container } = render(<BannerCollection />);

    expect(container).toMatchSnapshot();
  });
});
