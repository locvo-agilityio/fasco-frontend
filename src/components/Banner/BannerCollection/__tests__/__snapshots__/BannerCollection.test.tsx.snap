// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerCollection should render correctly 1`] = `
<div>
  <div
    class="relative flex w-full h-full max-h-[1077px] flex-col items-center justify-center bg-background-500"
  >
    <div
      class="flex w-full h-full"
    >
      <div
        class="hidden relative md:block w-1/2 h-full min-h-[570px]"
      >
        <img
          alt="Sign In Image"
          data-nimg="fill"
          decoding="async"
          sizes="50vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=100"
          srcset="/_next/image?url=%2Fimg.jpg&w=384&q=100 384w, /_next/image?url=%2Fimg.jpg&w=640&q=100 640w, /_next/image?url=%2Fimg.jpg&w=750&q=100 750w, /_next/image?url=%2Fimg.jpg&w=828&q=100 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=100 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=100 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=100 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=100 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=100 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <div
        class="flex flex-col w-full h-full md:w-1/2 items-start justify-center"
      >
        <div
          class="pt-14 px-14"
        >
          <p
            class="font-poppins text-gray-700"
          >
            Women Collection
          </p>
          <h2
            class="text-4xl font-volkhov text-primary-200 mt-5"
          >
            Peaky Blinders
          </h2>
          <p
            class="font-poppins underline mt-5"
          >
            DESCRIPTION
          </p>
          <p
            class="font-poppins text-gray-600 max-w-[515px] mt-5"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Scelerisque duis ultrices sollicitudin aliquam sem. Scelerisque duis ultrices sollicitudin. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Scelerisque duis.
          </p>
          <div
            class="mt-5 flex items-center gap-4"
          >
            Size:
            <div
              class="w-[53px] h-7 bg-black rounded-regular text-white text-center"
            >
              M
            </div>
          </div>
          <p
            class="font-poppins text-black font-medium mt-5 text-4md"
          >
            $100.00
          </p>
          <a
            class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none border-default data-[hover=true]:opacity-hover border-1 bg-primary-300 text-foreground-100 disabled:opacity-50 w-full py-5 text-base rounded-base data-[pressed=true]:scale-[1] font-poppins max-w-52 h-14 mt-10"
            href="/shop"
            role="button"
            tabindex="0"
          >
            Buy Now
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
